src/libraries/System.Net.Http/ref/System.Net.Http.cs

```diff
@@ -135,6 +135,8 @@ public partial class HttpClientHandler : System.Net.Http.HttpMessageHandler
         public long MaxRequestContentBufferSize { get { throw null; } set { } }
         [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
         public int MaxResponseHeadersLength { get { throw null; } set { } }
+        [System.CLSCompliantAttribute(false)]
+        public System.Diagnostics.Metrics.IMeterFactory? MeterFactory { get { throw null; } set { } }
```

Comment:
My personal hot-take is that we should put a big PREVIEW label on all the metrics features and API-s we are introducing in 8.0 because of all the uncertainties and the fact that [OTel semantic conventions standard](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#http-client) we are trying to conform to is also not stable yet.

> passing a delegate or an interface that could run some custom code

I really wish we can avoid that. It looks to me that all that we may need in the future is a name-mapping mechanism in `Meter` which could be implemented in many ways in `System.Diagnostics.Metrics`. Or is there more?

Reply:
> Or is there something I'm missing?

I am just anticipating that we may be asked for customizations that include new metrics, or things that are differently named, or the same names but differently measured. If the changes were non-breaking and apply to everyone then yes we could implement directly in HttpClient. But if it is people wanting customizations not suitable for everyone then we'd probably start looking at extensibility. I'm not advocating we add these things nor am I at all confident these requests will ever come, but I did want to put cards on the table that its at least a possibility.

> the fact that [OTel semantic conventions standard](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#http-client) we are trying to conform to is also not stable yet.

Based on recent discussions with Reiley they are stable (enough) and I am planning to put together a PR shortly that updates our names to match.